### PR TITLE
Support inline string

### PR DESCRIPTION
**Pull request by dplusic**
* Add support for inline strings.
* It is useful if you have a lot of distinct strings.
* References
  * [officeopenxml.com](http://officeopenxml.com/SScontentOverview.php#:~:text=s%3D%222%22%20t%3D%22-,inlineStr,-%22%3E)
  * [learn.microsoft.com](https://learn.microsoft.com/en-us/dotnet/api/documentformat.openxml.spreadsheet.cell?view=openxml-2.8.1#:~:text=t%3D%22-,inlineStr,-%22%3E%20%20%0A%20%20%20%20%3C)

### CODE REVIEW

File: fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java [hunk 1]
```diff
@@ -309,6 +309,7 @@ void testForAllFeatures() throws Exception {
 
             //set values for cell
             ws.value(0, 0, "Hello fastexcel");
+            ws.value(0, 1, "Hello fastexcel (inline)", true);
```
Comment (rzymek) [inline fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java:312]: I'm not sure about this API. Reading this line, it's hard to figure out what does the `true` stand for.

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/Cell.java [hunk 2]
```diff
@@ -52,11 +52,13 @@ void write(Writer w, int r, int c) throws IOException {
                 w.append(" s=\"").append(style).append('\"');
             }
             if (value != null && !(value instanceof Formula)) {
-                w.append(" t=\"").append((value instanceof CachedString) ? 's' : (value instanceof Boolean ? 'b' : 'n')).append('\"');
+                w.append(" t=\"").append((value instanceof CachedString) ? "s" : (value instanceof Boolean ? "b" : (value instanceof String ? "inlineStr" : "n"))).append('\"');
```
Comment (rzymek) [inline fastexcel-writer/src/main/java/org/dhatim/fastexcel/Cell.java:55]: This `?:` chain is getting too long. I suggest introducing a method call here:
```
w.append(" t=\"").append(getCellType(value)).append('\"');
```

File: fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java (hunk 1 above)
Comment (dplusic) [reply fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java:312]: I admit it.
Which would be better?

1. `ws.inlineString(0, 1, "Hello fastexcel (inline)");`
2. `ws.value(0, 1, "Hello fastexcel (inline)", StringStoreType.INLINE);`
3. `ws.value(0, 1, new InlineString("Hello fastexcel (inline)"));`

Or, could you suggest something else?

File: fastexcel-writer/src/main/java/org/dhatim/fastexcel/Cell.java (hunk 2 above)
Comment (dplusic) [reply fastexcel-writer/src/main/java/org/dhatim/fastexcel/Cell.java]: I added as suggested: https://github.com/dhatim/fastexcel/pull/266/commits/4125b8e0df8495a015cdffae16c9d910a41b1e04

File: fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java (hunk 1 above)
Comment (ochedru) [reply fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java:312]: +1 for `ws.inlineString`

Comment (dplusic) [reply fastexcel-writer/src/test/java/org/dhatim/fastexcel/CorrectnessTest.java]: @ochedru Thank you for your comment!

I changed it to `ws.inlineString`.

**Comment by ochedru** [comment]
Thank you for your PR! I merged it but we would need a few lines in the readme to describe this new feature. Can you handle this? Thanks.

**Comment by dplusic** [comment]
Thank you for merging this PR!
I'll add a description next to [the hyperlink section](https://github.com/dhatim/fastexcel#hyperlink).

**Pull request merged**